File: fizzbuzz_js/spec/fizzBuzz.spec.js
const subject = new FizzBuzz();

describe("FizzBuzz", () => {
  it("is expected to return a number is none of the game rules apply", () => {
    expect(subject.check(1)).toEqual(1)
  });

  it("is expected to return fizz if the number is divisible by 3", () => {
    expect(subject.check(3)).toEqual("fizz")
  });

  it("is expected to return buzz if the number is divisible by 5", () => {
    expect(subject.check(5)).toEqual("buzz")
  });

  it("is expected to return fizzbuzz if the number is divisible by 15", () => {
    expect(subject.check(15)).toEqual("fizz-buzz")
  });
});

File: fizzbuzz_rb/spec/fizz_buzz_spec.rb
require_relative '../src/fizz_buzz.rb'

describe FizzBuzz do

  subject { FizzBuzz.new }

  it 'is expected to return a number is none of the game rules apply' do
   expect(subject.check(2)).to eq 2 
  end

  it 'is expected to return fizz if the number is divisible by 3' do
    expect(subject.check(3)).to eq 'fizz'
  end
  
  it 'is expected to return buzz if the number is divisible by 5' do
    expect(subject.check(5)).to eq 'buzz'
  end 

  it 'is expected to return fizzbuzz if the number is divisible by 15' do
    expect(subject.check(15)).to eq 'fizz-buzz'    
  end
end